swarm/network/stream/snapshot_sync_test.go

```diff
@@ -138,230 +84,283 @@ func TestSyncing(t *testing.T) {
 		for _, chnk := range chnkCnt {
 			for _, n := range nodeCnt {
 				log.Info(fmt.Sprintf("Long running test with %d chunks and %d nodes...", chnk, n))
-				testSyncing(t, chnk, n)
+				testSyncingViaGlobalSync(t, chnk, n)
 			}
 		}
 	}
 }
 
-//Do run the tests
-//Every test runs 3 times, a live, a history, and a live AND history
-func testSyncing(t *testing.T, chunkCount int, nodeCount int) {
-	//test live and NO history
-	log.Info("Testing live and no history")
-	live = true
-	history = false
-	err := runSyncTest(chunkCount, nodeCount, live, history)
-	if err != nil {
-		t.Fatal(err)
-	}
-	//test history only
-	log.Info("Testing history only")
-	live = false
-	history = true
-	err = runSyncTest(chunkCount, nodeCount, live, history)
-	if err != nil {
-		t.Fatal(err)
-	}
-	//finally test live and history
-	log.Info("Testing live and history")
-	live = true
-	err = runSyncTest(chunkCount, nodeCount, live, history)
-	if err != nil {
-		t.Fatal(err)
+func TestSyncingViaDirectSubscribe(t *testing.T) {
+	//if nodes/chunks have been provided via commandline,
+	//run the tests with these values
+	if *nodes != 0 && *chunks != 0 {
+		log.Info(fmt.Sprintf("Running test with %d chunks and %d nodes...", *chunks, *nodes))
+		err := testSyncingViaDirectSubscribe(*chunks, *nodes)
+		if err != nil {
+			t.Fatal(err)
+		}
+	} else {
+		var nodeCnt []int
+		var chnkCnt []int
+		//if the `longrunning` flag has been provided
+		//run more test combinations
+		if *longrunning {
+			chnkCnt = []int{1, 8, 32, 256, 1024}
+			nodeCnt = []int{32, 16}
+		} else {
+			//default test
+			chnkCnt = []int{4, 32}
+			nodeCnt = []int{32, 16}
+		}
+		for _, chnk := range chnkCnt {
+			for _, n := range nodeCnt {
+				log.Info(fmt.Sprintf("Long running test with %d chunks and %d nodes...", chnk, n))
+				err := testSyncingViaDirectSubscribe(chnk, n)
+				if err != nil {
+					t.Fatal(err)
```

Comment:
Just once in 200 iterations, I got this error:

```
--- FAIL: TestSyncingViaDirectSubscribe (61.21s)
        snapshot_sync_test.go:120: context deadline exceeded
``` 

This test usually passes in around 15 seconds, so having the deadline at 60s is safe enough. This means that TestSyncingViaDirectSubscribe will probably be flaky even with longer deadline, but the chance for failure is extremely low by my results. I see no easy way to reproduce the problem and to debug it, so maybe just to leave it as it is.